libpod/container_inspect.go

```diff
@@ -0,0 +1,161 @@
+package libpod
+
+import (
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/projectatomic/libpod/libpod/driver"
+	"github.com/sirupsen/logrus"
+)
+
+func getContainerData(ctr *Container, size bool, driverData *driver.Data) (*ContainerData, error) {
+	config := ctr.config
+	runtimeInfo := ctr.state
+	spec := ctr.config.Spec
+
+	cpus, mems, period, quota, realtimePeriod, realtimeRuntime, shares := getCPUInfo(spec)
+	blkioWeight, blkioWeightDevice, blkioReadBps, blkioWriteBps, blkioReadIOPS, blkioeWriteIOPS := getBLKIOInfo(spec)
+	memKernel, memReservation, memSwap, memSwappiness, memDisableOOMKiller := getMemoryInfo(spec)
+	pidsLimit := getPidsInfo(spec)
+
+	args := config.Spec.Process.Args
+	path := args[0]
+	if len(args) > 0 {
+		args = args[1:]
+	}
+
+	data := &ContainerData{
+		Id:      config.ID,
+		Created: config.CreatedTime,
+		Path:    path,
+		Args:    args,
+		State: &CtrState{
+			OciVersion: spec.Version,
+			Status:     runtimeInfo.State.String(),
+			Running:    runtimeInfo.State == ContainerStateRunning,
+			Paused:     runtimeInfo.State == ContainerStatePaused,
+			OOMKilled:  runtimeInfo.OOMKilled,
+			Dead:       runtimeInfo.State.String() == "bad state",
+			Pid:        runtimeInfo.PID,
+			ExitCode:   runtimeInfo.ExitCode,
+			Error:      "", // can't get yet
+			StartedAt:  runtimeInfo.StartedTime,
+			FinishedAt: runtimeInfo.FinishedTime,
+		},
+		ImageID:         config.RootfsImageID,
+		ImageName:       config.RootfsImageName,
+		ResolvConfPath:  runtimeInfo.ConfigPath,                               // not sure
+		HostnamePath:    spec.Annotations["io.kubernetes.cri-o.HostnamePath"], // not sure
+		HostsPath:       "",                                                   // can't get yet
+		StaticDir:       config.StaticDir,
+		LogPath:         ctr.LogPath(),
+		Name:            config.Name,
+		Driver:          driverData.Name,
+		MountLabel:      config.MountLabel,
+		ProcessLabel:    spec.Process.SelinuxLabel,
+		AppArmorProfile: spec.Process.ApparmorProfile,
+		ExecIDs:         "", //can't get yet
+		HostConfig: &HostConfig{
+			Binds:                config.Volumes,
```

Comment:
I really don't like storing this in Container. It doesn't directly affect any other part of the container, because the mounts are already in the spec. If possible, we should parse them out of the spec. Otherwise, we should either throw them in a label, or do something similar to the Artifacts functions we discussed on IRC.

Reply:
Thinking about it more, we can't tell what the default mounts are, so even if it's not completely compliant with the Docker CLI we should just output all mounts in the spec.

Reply:
If we already include the mounts (I see them below), I would probably just remove this and the container.config.Volumes field. From a libpod standpoint we don't know what is a user-created bind mount and what isn't - we just know about the mounts in the spec.

Reply:
removed volumes and binds. fixed